### PR TITLE
Add CI via GitHub actions.

### PR DESCRIPTION
**Pull request by ericonr**
Arch and Void containers, initially.

### CODE REVIEW

**Comment by ericonr** [comment]
Huh, I hadn't seen #32, sorry. Feel free to close this one, though I'd still like to get a Void CI run, and I believe the approach of doing it all in a single YAML file is a bit simpler.

**Comment by Foxboron** [comment]
Hmm, I want the `Makefile`, `README` and code fixups from #32 and the CI file from this PR. Can we maybe split up the first one and rebase this on that work?

Thoughts @syscll?

**Comment by syscll** [comment]
Sounds good to me!

**Comment by ericonr** [comment]
@syscll can you open a new PR / update #32 to not include the CI changes?

I can update this once that's done.

**Comment by Foxboron** [comment]
Feel free to squash the changes as well :)

**Comment by syscll** [comment]
Have updated #32 to remove CI and Dockerfile.

**Comment by Foxboron** [comment]
@ericonr Please rebase on master and include `make lint` and `make test`

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: CI
+on: [push, pull_request]
+jobs:
+   arch:
+      runs-on: ubuntu-latest
+      container: archlinux:latest
+      steps:
+         - run: pacman --noconfirm --noprogressbar -Syyu
+         - run: pacman --noconfirm --noprogressbar -Sy make go asciidoc
```
Comment (Foxboron) [inline .github/workflows/ci.yaml:9]: `pacman -S` should be used given the `-Syyu` on the previous line which can be `-Syu`.

**Comment by ericonr** [comment]
@Foxboron I think you need to enable something in the repo for actions to work; it should be running a check in the PR, I think.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (ericonr) [reply .github/workflows/ci.yaml]: Fixed.

**Comment by Foxboron** [comment]
Lets merge it and see what happens :)

**Comment by ericonr** [comment]
I needed to add gcc for the tests to run. You can see progress in https://github.com/ericonr/sbctl/actions

**Comment by Foxboron** [comment]
Still failing :D I'll merge this and fix in master :) Thanks for the work!

**Comment by ericonr** [comment]
Oops, I just pushed a fix: `         - run: PATH=$PATH:$HOME/.cache/go/bin make deps lint` might be enough

**Comment by ericonr** [comment]
And it wasn't. Might be easier to run `make deps` separately with an explicit `GOBIN`.

**Pull request merged**